src/main/kotlin/com/mdrobnak/lalrpop/psi/ext/LpAction.kt

```diff
@@ -111,7 +111,8 @@ abstract class LpActionMixin(node: ASTNode) : ASTWrapperPsiElement(node), LpActi
 
         val code = actionCodeEscape(code.text, alternativeParent.selectedTypesInContext(context))
 
-        val genericUnitStructs = alternativeParent.nonterminalParent.rustGenericUnitStructs()
+        val grammarDecl = containingFile.lalrpopFindGrammarDecl()
+        val genericUnitStructs = grammarDecl.typeParamsRustUnitStructs() + arguments.rootArguments.joinToString { "struct ${it.name};\n" }
```

Comment:
I think `joinToString` still has the default separator of `", "`? That might cause problems.